api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/MIReporter.java

```diff
@@ -1,12 +1,68 @@
 package uk.gov.dhsc.htbhf.claimant.reporting;
 
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
 import uk.gov.dhsc.htbhf.claimant.entity.Claim;
+import uk.gov.dhsc.htbhf.claimant.exception.PostcodesIoClientException;
+import uk.gov.dhsc.htbhf.claimant.model.PostcodeData;
+import uk.gov.dhsc.htbhf.claimant.model.PostcodeDataResponse;
+import uk.gov.dhsc.htbhf.claimant.repository.ClaimRepository;
 
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+/**
+ * Responsible for reporting 'events' to Google Analytics (events including new claims and payments made, etc).
+ * Also responsible for obtaining postcode data used to provide location info when reporting events.
+ */
 @Component
+@Slf4j
 public class MIReporter {
 
+    private static final String POSTCODES_PATH = "/postcodes/";
+
+    private final RestTemplate restTemplate;
+    private final ClaimRepository claimRepository;
+    private final String postcodesUri;
+
+    public MIReporter(@Value("${postcodes-io.base-uri}") String postcodesBaseUri,
+                      RestTemplate restTemplate,
+                      ClaimRepository claimRepository) {
+        this.restTemplate = restTemplate;
+        this.claimRepository = claimRepository;
+        this.postcodesUri = postcodesBaseUri + POSTCODES_PATH;
+    }
+
     public void reportClaim(Claim claim) {
-        // TODO DW HTBHF-2418 Populate the claim with post code data and report claim to google analytics
+        if (claim.getPostcodeData() == null) {
+            PostcodeData postcodeData = getPostcodeData(claim);
+            claim.setPostcodeData(postcodeData);
+            claimRepository.save(claim);
+        }
+    }
+
+    private PostcodeData getPostcodeData(Claim claim) {
+        String postcode = claim.getClaimant().getAddress().getPostcode().replace(" ", "");
```

Comment:
Is this worth being in its own method so you can explain why we need to do this? I presume we're doing this so we don't have a space in the built URL?

Reply:
Moved into it's own method to be clearer.